lib/ask/runtime/channel_broker_agent.ex

```diff
@@ -27,7 +27,7 @@ defmodule Ask.Runtime.ChannelBrokerAgent do
 
   def save_state(%State{channel_id: channel_id} = state) do
     Agent.update(__MODULE__, fn agent ->
-      Map.put(agent, channel_id, state)
+      Map.put(agent, channel_id, Map.delete(state, :runtime_channel))
```

Comment:
Asking here again because is not clear to me: this new implementation does not do the same as before and it is not clear to me why we want to remove the runtime channel now (and not before).

See for example `ChannelBroker.refresh_runtime_channel(state)`

We return (when channel is about to expire):

```
state
      |> Map.put(:runtime_channel, new_runtime_channel)
      |> Agent.save_state()
```

We put the runtime channel in the state and then we remove it?

Reply:
Urgh, you got a bug! I only want to remove it from the saved state in the agent, so that when we restore the state, we create a new one.

But the `ChannelBrokerAgent.save_state/1` method should return the pristine `state` object, indeed. Right now we probably reload the runtime_channel _every_ call to the ChannelBroker :facepalm:

Reply:
@leandroradusky Ah no, that's OK: the method returns the original `state`. So I answered your question above: no need to save the `runtime_channel` it's not important, and in case of a channel-broker crash we only need to restore the active/queued contacts, and we can reset the rest.